svgnative/example/testCairo/TestCairo.cpp

```diff
@@ -0,0 +1,184 @@
+
+    return true;
+#endif
+}
+
+bool try_emit_cairo_script(const char* pathCairoScript, cairo_surface_t* cairoRecSurface)
+{
+    assert( pathCairoScript );
+    assert( cairoRecSurface );
+
+#ifndef CAIRO_HAS_SCRIPT_SURFACE
+    return false;
+#else
+    auto cairoScriptDev = cairo_script_create(pathCairoScript);
+    cairo_script_from_recording_surface(cairoScriptDev, cairoRecSurface);
+    shutdown_cairo_device( cairoScriptDev );
+
+    return true;
+#endif
+}
+
+bool try_emit_png(const char* pathPng, cairo_surface_t* cairoRecSurface)
+{
+    assert( pathPng );
+    assert( cairoRecSurface );
+
+#ifndef CAIRO_HAS_PNG_FUNCTIONS
+    // TODO: rather than using shortcut PNG function, getting the pixel data and
+    //       pack it in specified image format would be better to support various
+    //       raster image formats, like JPEG and GIF.
+
+    return false;
+#else
+    cairo_surface_write_to_png( cairoRecSurface, pathPng );   
+
+    return true;
+#endif
+}
+
+int main(int argc, char* const argv[])
+{
+    if (argc < 3)
+    {
+        std::cerr << argv[0] << " <input.svg> <output1.png> [<output2.xml> <output3.cs> <output4.svg> ...]" << std::endl;
+        return 0;
+    }
+
+    std::string svgInput{};
+    std::ifstream input(argv[1]);
+    if (!input)
+    {
+        std::cerr << "Error! Could not open input file." << std::endl;
+        exit(EXIT_FAILURE);
+    }
+    for (std::string line; std::getline(input, line);)
+        svgInput.append(line);
+    input.close();
+
+    auto renderer = std::make_shared<SVGNative::CairoSVGRenderer>();
+
+    auto doc = std::unique_ptr<SVGNative::SVGDocument>(SVGNative::SVGDocument::CreateSVGDocument(svgInput.c_str(), renderer));
+
+    cairo_rectangle_t docExtents = { 0, 0, 0, 0 };
+    docExtents.width = doc->Width();
+    docExtents.height = doc->Height();
+
+    for (int i = 2; i < argc; i++ )
+    {
+        std::string outPath = argv[i];
+        std::string suffix = outPath.substr(outPath.rfind('.') + 1).c_str();
+        std::transform(suffix.begin(), suffix.end(), suffix.begin(), ::tolower);
+
+        cairo_surface_t* cairoRecSurface = cairo_recording_surface_create( CAIRO_CONTENT_COLOR_ALPHA, &docExtents );
+
+        auto cairoRecContext = cairo_create( cairoRecSurface );
+
+        renderer->SetCairo( cairoRecContext );
+        doc->Render();
+        cairo_destroy( cairoRecContext );
+
+        if (suffix == "svg" && try_emit_svg(outPath.c_str(), doc->Width(), doc->Height(), cairoRecSurface))
+            continue;
+
+        if (suffix == "xml" && try_emit_xml(outPath.c_str(), cairoRecSurface))
+            continue;
+
+        if (suffix == "cs" && try_emit_cairo_script(outPath.c_str(), cairoRecSurface))
+            continue;
+
+        if (suffix == "png" && try_emit_png(outPath.c_str(), cairoRecSurface))
+            continue;
+
+        std::cerr << "Specified format " << suffix << " does not match with any output format available in linked Cairo" << std::endl;
```

Comment:
s/does not match with any output format/does not match any output format/

Reply:
Thanks, and sorry for my poor English!